RecoBTag/DeepFlavour/interface/deep_helpers.h

```diff
@@ -0,0 +1,108 @@
+#ifndef RecoBTag_DeepFlavour_deep_helpers_h
+#define RecoBTag_DeepFlavour_deep_helpers_h
+
+#include "DataFormats/GeometryCommonDetAlgo/interface/Measurement1D.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "TrackingTools/IPTools/interface/IPTools.h"
+
+
+
+namespace deep {
+
+  // adapted from DeepNTuples
+  inline const float& catch_infs(const float& in,
+                                 const float& replace_value) {
+    if(in==in){ // why is this ?
+      if(std::isinf(in))
+        return replace_value;
+      else if(in < -1e32 || in > 1e32)
+        return replace_value;
+      return in;
+    }
+    return replace_value;
+  } 
+
+  // adapted from DeepNTuples
+  inline const float catch_infs_and_bound(const float& in,
+                                          const float& replace_value,
+                                          const float& lowerbound,
+                                          const float& upperbound,
+                                          const float offset=0.,
+                                          const bool& use_offsets = true){
+        float withoutinfs=catch_infs(in,replace_value);
+        if(withoutinfs+offset<lowerbound) return lowerbound;
+        if(withoutinfs+offset>upperbound) return upperbound;
+        if(use_offsets)
+            withoutinfs+=offset;
+        return withoutinfs;
+  }
+
+
+  // adapted from DeepNTuples 
+  Measurement1D vertexDxy(const reco::VertexCompositePtrCandidate &svcand, const reco::Vertex &pv)  {
+    VertexDistanceXY dist;
+    reco::Vertex::CovarianceMatrix csv; svcand.fillVertexCovariance(csv);
+    reco::Vertex svtx(svcand.vertex(), csv);
+    return dist.distance(svtx, pv);
+  }
+
+  // adapted from DeepNTuples 
+  Measurement1D vertexD3d(const reco::VertexCompositePtrCandidate &svcand, const reco::Vertex &pv)  {
+    VertexDistance3D dist;
+    reco::Vertex::CovarianceMatrix csv; svcand.fillVertexCovariance(csv);
+    reco::Vertex svtx(svcand.vertex(), csv);
+    return dist.distance(svtx, pv);
+  }
+
+  // adapted from DeepNTuples 
+  float vertexDdotP(const reco::VertexCompositePtrCandidate &sv, const reco::Vertex &pv)  {
+    reco::Candidate::Vector p = sv.momentum();
+    reco::Candidate::Vector d(sv.vx() - pv.x(), sv.vy() - pv.y(), sv.vz() - pv.z());
+    return p.Unit().Dot(d.Unit());
+  }
+
+  // adapted from DeepNTuples 
```

Comment:
this isn't much of a documentation for this method.
Please add some more explicit notes if the goal was to provide doc

Reply:
will do

Reply:
added in https://github.com/cms-sw/cmssw/pull/19893/commits/1d9e7d0e37beede555eb8e48850dfbcf4141bcb0